# 🛒 Superstore Sales Analysis

## Data Cleaning & Visualization using Python

### Author
**Sarthak Paturkar**

### Objective
Analyze the Superstore Sales dataset by cleaning the data, exploring trends, creating visualizations, and generating business insights.

### Tools Used
- Python
- Pandas
- NumPy
- Matplotlib
- Seaborn
- VS Code + Jupyter Notebook

In [10]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [11]:
df = pd.read_csv("../data/samplesuperstore.csv")

In [12]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country/Region,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,US-2023-103800,1/3/2023,1/7/2023,Standard Class,DP-13000,Darren Powers,Consumer,United States,Houston,...,77095,Central,OFF-PA-10000174,Office Supplies,Paper,"Message Book, Wirebound, Four 5 1/2"" X 4"" Form...",16.448,2,0.2,5.5512
1,2,US-2023-112326,1/4/2023,1/8/2023,Standard Class,PO-19195,Phillina Ober,Home Office,United States,Naperville,...,60540,Central,OFF-BI-10004094,Office Supplies,Binders,GBC Standard Plastic Binding Systems Combs,3.540,2,0.8,-5.4870
2,3,US-2023-112326,1/4/2023,1/8/2023,Standard Class,PO-19195,Phillina Ober,Home Office,United States,Naperville,...,60540,Central,OFF-LA-10003223,Office Supplies,Labels,Avery 508,11.784,3,0.2,4.2717
3,4,US-2023-112326,1/4/2023,1/8/2023,Standard Class,PO-19195,Phillina Ober,Home Office,United States,Naperville,...,60540,Central,OFF-ST-10002743,Office Supplies,Storage,SAFCO Boltless Steel Shelving,272.736,3,0.2,-64.7748
4,5,US-2023-141817,1/5/2023,1/12/2023,Standard Class,MB-18085,Mick Brown,Consumer,United States,Philadelphia,...,19143,East,OFF-AR-10003478,Office Supplies,Art,Avery Hi-Liter EverBold Pen Style Fluorescent ...,19.536,3,0.2,4.8840


In [13]:
df.shape

(10194, 21)

### Dataset Shape

The dataset contains **10,194 rows** and **21 columns**.

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10194 entries, 0 to 10193
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Row ID          10194 non-null  int64  
 1   Order ID        10194 non-null  object 
 2   Order Date      10194 non-null  object 
 3   Ship Date       10194 non-null  object 
 4   Ship Mode       10194 non-null  object 
 5   Customer ID     10194 non-null  object 
 6   Customer Name   10194 non-null  object 
 7   Segment         10194 non-null  object 
 8   Country/Region  10194 non-null  object 
 9   City            10194 non-null  object 
 10  State/Province  10194 non-null  object 
 11  Postal Code     10194 non-null  object 
 12  Region          10194 non-null  object 
 13  Product ID      10194 non-null  object 
 14  Category        10194 non-null  object 
 15  Sub-Category    10194 non-null  object 
 16  Product Name    10194 non-null  object 
 17  Sales           10194 non-null 

## Dataset Information

### Observations

- The dataset contains **10,194 rows** and **21 columns**.
- There are **no missing values** in any column.
- The dataset contains:
  - **16 object (text) columns**
  - **2 integer columns**
  - **3 float columns**
- The **Order Date** and **Ship Date** columns are currently stored as **object (text)** and should be converted to **datetime** format during data cleaning.
- The dataset uses approximately **1.6 MB** of memory.

### Conclusion

The dataset is complete and does not contain any missing values. However, the date columns need to be converted to the correct data type before performing further analysis.

In [15]:
df.describe()

,Row ID,Sales,Quantity,Discount,Profit
count,10194.000000,10194.000000,10194.000000,10194.000000,10194.000000
mean,5097.500000,228.225854,3.791838,0.155385,28.673417
std,2942.898656,619.906839,2.228317,0.206249,232.465115
min,1.000000,0.444000,1.000000,0.000000,-6599.978000
25%,2549.250000,17.220000,2.000000,0.000000,1.760800
50%,5097.500000,53.910000,3.000000,0.200000,8.690000
75%,7645.750000,209.500000,5.000000,0.200000,29.297925
max,10194.000000,22638.480000,14.000000,0.800000,8399.976000


## Statistical Summary

### Observations

- The statistical summary is generated for the numerical columns:
  - **Row ID**
  - **Sales**
  - **Quantity**
  - **Discount**
  - **Profit**

- The dataset contains **10,194 records**, confirming there are no missing values in the numerical columns.

- The **average sales** per order is **228.23**.

- Customers purchase an average of **3.79 products** per order.

- The **average discount** offered is **15.54%**.

- The **average profit** earned per order is **28.67**.

- The **minimum sales** recorded is **0.44**, while the **maximum sales** recorded is **22,638.48**.

- The **minimum profit** is **-6,599.98**, indicating that some orders resulted in significant losses.

- The **maximum profit** is **8,399.98**, showing that some orders generated exceptionally high profits.

- The large difference between the minimum and maximum values suggests the presence of **outliers**, especially in the **Sales** and **Profit** columns.

### Conclusion

The statistical summary provides a clear overview of the numerical data. The wide variation in **Sales** and **Profit** indicates that outlier detection will be an important step during data cleaning. Overall, the dataset is complete and suitable for further exploratory data analysis.

In [16]:
df.isnull().sum()

Row ID            0
Order ID          0
Order Date        0
Ship Date         0
Ship Mode         0
Customer ID       0
Customer Name     0
Segment           0
Country/Region    0
City              0
State/Province    0
Postal Code       0
Region            0
Product ID        0
Category          0
Sub-Category      0
Product Name      0
Sales             0
Quantity          0
Discount          0
Profit            0
dtype: int64

## Missing Value Analysis

### Observations

- The dataset was checked for missing values using `df.isnull().sum()`.
- All **21 columns** contain **0 missing values**.
- Every record is complete, and no values are missing in any feature.

### Conclusion

Since the dataset does not contain any missing values, no imputation or deletion of records is required. The dataset is complete and ready for the next data cleaning step.

In [17]:
df.duplicated().sum()

np.int64(0)

## Duplicate Value Analysis

### Observations

- The dataset was checked for duplicate records using `df.duplicated().sum()`.
- The output returned **0**, which indicates that there are no duplicate rows in the dataset.
- Every record in the dataset is unique.

### Conclusion

No duplicate records were found. Therefore, no rows need to be removed, and the dataset is ready for the next stage of data cleaning and analysis.

In [18]:
df.dtypes

Row ID              int64
Order ID           object
Order Date         object
Ship Date          object
Ship Mode          object
Customer ID        object
Customer Name      object
Segment            object
Country/Region     object
City               object
State/Province     object
Postal Code        object
Region             object
Product ID         object
Category           object
Sub-Category       object
Product Name       object
Sales             float64
Quantity            int64
Discount          float64
Profit            float64
dtype: object

## Data Type Analysis

### Observations

- The data types of all columns were examined using `df.dtypes`.
- Most categorical columns are stored as **object** data type.
- The numerical columns such as **Sales**, **Quantity**, **Discount**, and **Profit** have appropriate numeric data types.
- The **Order Date** and **Ship Date** columns are currently stored as **object (text)** instead of **datetime**.

### Conclusion

The date columns need to be converted into the **datetime** format before performing time-based analysis such as monthly sales trends and shipping duration.

In [19]:
df["Order Date"] = pd.to_datetime(df["Order Date"])
df["Ship Date"] = pd.to_datetime(df["Ship Date"])

In [20]:
df.dtypes

Row ID                     int64
Order ID                  object
Order Date        datetime64[ns]
Ship Date         datetime64[ns]
Ship Mode                 object
Customer ID               object
Customer Name             object
Segment                   object
Country/Region            object
City                      object
State/Province            object
Postal Code               object
Region                    object
Product ID                object
Category                  object
Sub-Category              object
Product Name              object
Sales                    float64
Quantity                   int64
Discount                 float64
Profit                   float64
dtype: object

## Date Conversion

### Observations

- The **Order Date** and **Ship Date** columns were successfully converted from **object** to **datetime64[ns]**.
- The dataset is now ready for time-based analysis such as monthly sales, yearly trends, and shipping duration calculations.

### Conclusion

Converting date columns into the correct format improves data accuracy and enables advanced date-related analysis.